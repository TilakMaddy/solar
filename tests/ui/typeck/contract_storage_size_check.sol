struct Person {
    string name;
    uint age;
}

contract B {
    uint c;
    bool x;
}

// Total = 1 + 1 = 2

contract A {
    uint256 a; // 1
    bool b; // 1
    B c; // 1
    Person e; // 1 + 2 fields = 3
    int128 f; // 1
    Person[] g; // 1
    Person[23] h; // 23 * 3 = 69
}

// Total = 1 + 1 + 1 + 3 + 1 + 1 + 69 = 77

contract M {
    struct P1 {
        string first;
        string middle;
        string last;
    }

    P1 my; // 4
    mapping(string => uint256) public a; // 1
    P1[] public b; // 1
    bool c; // 1
    B d; // 1
}

// Total = 4 + 1 + 1 + 1 + 1 = 8
