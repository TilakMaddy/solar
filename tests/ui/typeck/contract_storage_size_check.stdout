ROOT/tests/ui/typeck/contract_storage_size_check.sol:B requires a maximum of 2 storage slots
ROOT/tests/ui/typeck/contract_storage_size_check.sol:A requires a maximum of 77 storage slots
ROOT/tests/ui/typeck/contract_storage_size_check.sol:M requires a maximum of 8 storage slots
